tests/snippets/stdlib_os.py

```diff
@@ -63,27 +63,37 @@
 assert_raises(TypeError, lambda: os.fspath([1,2,3]))
 
 class TestWithTempDir():
-	def __enter__(self):
-		if os.name == "nt":
-			base_folder = os.environ["TEMP"]
-		else:
-			base_folder = "/tmp"
-		name = os.path.join(base_folder, "rustpython_test_os_" + str(int(time.time())))
-		os.mkdir(name)
-		self.name = name
-		return name
-
-	def __exit__(self, exc_type, exc_val, exc_tb):
-		# TODO: Delete temp dir
-		pass
+    def __enter__(self):
+        if os.name == "nt":
+            base_folder = os.environ["TEMP"]
+        else:
+            base_folder = "/tmp"
+
+        name = os.path.join(base_folder, "rustpython_test_os_" + str(int(time.time())))
+        os.mkdir(name)
+        self.name = name
+        return name
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        for root, dirs, files in os.walk(self.name, topdown=False):
+            for name in files:
+                os.remove(os.path.join(root, name))
+            for name in dirs:
+                to_remove = os.path.join(root, name)
+                if os.path.islink(to_remove):
+                    os.unlink(to_remove)
+                else:
+                    os.rmdir(to_remove)
```

Comment:
This will fail if the directory contains files in it. I think keeping the temporary folder for now is OK.

Reply:
The problem with keeping this directory is that the timestamp does not have enough time to get incremented during executing that tests - and simply tests fail.

I'll remove this ```__exit__``` code and try some manipulating with dirname.